istio-control/istio-discovery/files/injection-template.yaml

```diff
@@ -0,0 +1,373 @@
+# Configmap optimized for Istiod. Please DO NOT MERGE all changes from istio - in particular those dependent on
+# Values.yaml, which should not be used by istiod.
+
+# Istiod only uses SDS based config ( files will mapped/handled by SDS).
```

Comment:
this file is reverting a few fixes that landed. Looks like the intent was to remove just SDS and add CA_ADDR but it did more?

Reply:
This is a little concerning. We should try to find a way to share between istio and istiod.  Maybe istio could read this file and patch it?

Reply:
The intent is to simplify - we want the istiod config to be as easy as possible, and readable by admins. And certainly not using any values.yaml. 

Sometimes 'sharing' implies complexity, and not sharing is easier. 

Note that we also want to support users bringing their own (clean) injection templates. We know users who have this use case ( like Google, or others not using envoy, etc ). This will imply some stability in the injector API surface and making the template as minimal as possible - the work on Pilot returning mesh.config and the bootstrap is part of it.